Source/Core/VideoCommon/Fifo.cpp

```diff
@@ -463,10 +478,13 @@ static int RunGpuOnCpu(int ticks)
 		FPURoundMode::LoadSIMDState();
 	}
 
-	available_ticks = std::min(available_ticks, 0);
-	s_sync_ticks.store(available_ticks);
+	s_sync_ticks.store(std::min(available_ticks, 0));
+
+	if(!(fifo.bFF_GPReadEnable && fifo.CPReadWriteDistance && !AtBreakpoint()))
+		return -1;
 
-	return 10000 - available_ticks;
+	// always wait at least 100 cycles
+	return std::max(-available_ticks, 1000);
```

Comment:
Make the comment agree with the constant. (either 100 or 1000 is fine they are just magic numbers)